### PR TITLE
Guirong taking over for Luis - Unit test for getAllInvInProjectWBS inside InventoryController

### PR DESCRIPTION
**Pull request by luisarevalo21**
# Description
Unit testing for Inventory Controller

## Related PRS (if any):
N/A

## Main changes explained:
First set of unit test for the Inventory Controller 
Created unit tests for getAllInvInProjectWBS function
Created a md file describing all positive and negative cases. 

## How to test:
1. check into current branch
2. do` npm install`
3. run `npm run test inventoryController`
4. verify all tests pass
## Note:

Screenshot 2024-05-12 at 7 51 44 PM

### CODE REVIEW

**Review by j-corp-25 — APPROVED** [review]
Hi @luisarevalo21 ,

All tests are passing!


PR#945

**Review by Ramaruva — APPROVED** [review]
Hi I have run all test cases in my local env and are working as expected 
image

**Comment by KaushikMreddy** [comment]
Hi, 
All the test cases have passed
image

Please remove the commented code in the file, if your changes are final. Good work!

**Review by gowthamdongari — APPROVED** [review]
works as exxpected 
image

**Review by MinSun0814 — APPROVED** [review]
image
Hello, I ran your tests and everything passed.

File: src/controllers/inventoryController.spec.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        .spyOn(sortedObject, 'sort')
+        .mockImplementationOnce(() => Promise.reject(new Error('an error occured')));
+
+      const response = await getAllInvInProjectWBS(mockReq, mockRes);
+
+      expect(hasPermissionSpy).toHaveBeenCalledWith(
+        mockReq.body.requestor,
+        'getAllInvInProjectWBS',
+      );
+      expect(findSpy).toHaveBeenCalledWith({
```
Comment (Abi-Liu) [inline src/controllers/inventoryController.spec.js:77]: You do a good job of checking to make sure these methods are called with the correct arguments here.

File: src/controllers/inventoryController.spec.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        select: () => {},
+        lean: () => {},
+      };
+
+      const wbsExists = {
+        select: () => {},
+        lean: () => {},
+      };
+
+      const findSpyProjects = jest.spyOn(projects, 'findOne').mockImplementationOnce(() => {
```
Comment (Abi-Liu) [inline src/controllers/inventoryController.spec.js:155]: Make sure you assert that these spy methods are called with the correct arguments as well.

**Review by Abi-Liu — CHANGES_REQUESTED** [review]
Looks good! All test cases are passing as expected. I left a few comments regarding how to enhance the overall tests.
Screenshot 2024-05-17 at 7 37 42 PM

**Review by Wenboooooo — APPROVED** [review]
Looks good to me. All cases have been passed

**Review by aaryaneil — APPROVED** [review]
All the test cases passed.
Screenshot 2024-05-18 at 2 32 54 PM

**Review by dikshitakejriwal — APPROVED** [review]
All the test cases passed!
Screenshot 2024-05-31 at 12 54 28 PM

**Review by Sandhya1236 — APPROVED** [review]
I have successfully tested the first set of unit tests for the Inventory Controller. The main changes include creating unit tests for the getAllInvInProjectWBS function and documenting all positive and negative cases in a markdown file. After checking into the current branch, running npm install, and executing npm run test inventoryController, I verified that all tests passed successfully, confirming the functionality of the getAllInvInProjectWBS function is working as expected.

image

**Review by deepthikannan — APPROVED** [review]
image
Great code, all test cases passed !

**Comment by akshit0211** [comment]
All test cases passed. Good job!

All test cases passed

**Review by akshit0211 — APPROVED** [review]

**Review by shradac — APPROVED** [review]
All tests passed. The tests cover all specified scenarios. Good Work!

**Comment by mianmiantea2019** [comment]
All tests passed. The tests cover all specified scenarios. Good Work!

image

**Review by nyelmela — APPROVED** [review]
All test cases have passed.
image

**Comment by cecilia-uu** [comment]
截屏2024-07-21 下午3 16 04
Tests have passed.

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and all the test cases have passed.
Screenshot (159)

**Review by jinxyou — APPROVED** [review]
All test cases have passed.
PR 945

**Comment by Gmon-k** [comment]
All the test cases has passed and Below is the screenshot for it.

**Review by AJAYINAVOLU — APPROVED** [review]
Tests are passing successfully without any errors.

image

Thank You

**Review by manikittu810 — APPROVED** [review]
Hello @luisarevalo21 
Everything is working good as mentioned.

Thank you.
Screenshot 2024-09-05 at 7 16 40 PM

**Review by Saniya98 — APPROVED** [review]
All tests are passing 
image

**Review by neeharikakoniki — APPROVED** [review]
All tc passed.
image

**Review by panchalvatneu — APPROVED** [review]
All tests on inventoryController passed, good work!
image

**Review by ishangoel17 — COMMENTED** [review]
All test cases are passing. Below is the screenshot for reference:
Capture945

**Review by anirudhsk2107 — APPROVED** [review]
The tests are working properly 

Screen Shot 2024-09-24 at 2 44 36 PM

**Comment by geetamatkar** [comment]
I tested this PR and all test cases are passing.

PR945

**Review by ziyanw52 — APPROVED** [review]
I have tested that all the test cases have passed.
1191728197303_ pic_hd

**Comment by mianmiantea2019** [comment]
I have tested that all the test cases have passed.

image

**Comment by pallavithorat** [comment]
All test cases have passed.
PR#945

**Review by RachanaZha — APPROVED** [review]
Have tested the changes. All test cases passed successfully. Changes are working fine.

zha_rachana_PR#945

**Comment by SatyashanthiT** [comment]
I reviewed this pr for the Inventory Controller, specifically focusing on the unit tests for getAllInvInProjectWBS. After running the tests and verifying all cases pass, I approved this PR
image

**Comment by one-community** [comment]
Thank you all, merging!

**Comment by one-community** [comment]
Thank you all, merging!
